specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json

```diff
@@ -410,6 +410,86 @@
         "deprecated": false
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.IoTCentral/applicationTemplates/{applicationTemplateId}": {
+      "get": {
+        "tags": [
+          "Apps"
+        ],
+        "description": "Get a single application template.",
+        "operationId": "Apps_Template",
+        "x-ms-examples": {
+          "Apps_Template": {
+            "$ref": "./examples/Apps_Template.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/applicationTemplateId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The body contains the filtered IoT Central application template.",
+            "schema": {
+              "$ref": "#/definitions/AppTemplate"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.IoTCentral/applicationTemplates": {
+      "get": {
+        "tags": [
+          "Apps"
+        ],
+        "description": "Get all available application templates.",
```

Comment:
>application templates [](start = 42, length = 21)

Is "application template" an official name for this IOT central feature? Could a user confuse "application template" with ARM templates easily?

Reply:
changed it to IoTAppTemplates

Reply:
Sorry for the confusion. I was just asking the question rather than making a suggestion to change the naming. "IoTAppTemplates" actually violates the naming rules because we require camelCasing. I'm ok with either "appTemplates" or "applicationTemplates".

---
In reply to: [271079702](https://github.com/Azure/azure-rest-api-specs/pull/5495#discussion_r271079702) [](ancestors = 271079702)

Reply:
So would this name "IoTAppTemplates" work? we discuss it out

Reply:
Yeah iotAppTemplates would be ok too. (ARM resource IDs are expected to be case-insensitive, so your service should accept all variations.)

Reply:
As we have already have Microsoft.IoTCentral as the namespace, we decide to keep the url as appTemplates. Thanks